src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2337,5 +2341,44 @@ private IInstanceReferenceOperation CreateCollectionValuePlaceholderOperation(Bo
             bool isImplicit = placeholder.WasCompilerGenerated;
             return new InstanceReferenceOperation(referenceKind, _semanticModel, syntax, type, isImplicit);
         }
+
+        private ImmutableArray<IArgumentOperation> CreateDisposeArguments(MethodSymbol? disposeMethod, TypeSymbol? receiverType, SyntaxNode receiverSyntax, Binder binder)
+        {
+            if (disposeMethod is null || receiverType is null || disposeMethod.ParameterCount == 0)
+            {
+                return ImmutableArray<IArgumentOperation>.Empty;
+            }
+
+            var receiver = new BoundDisposableValuePlaceholder(receiverSyntax, receiverType);
+
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(disposeMethod.ParameterCount);
+            ImmutableArray<int> argsToParams = default;
+            var expanded = disposeMethod.HasParamsParameter();
+
+            binder.BindDefaultArguments(
+                receiver.Syntax,
+                disposeMethod.Parameters,
+                argumentsBuilder,
+                argumentRefKindsBuilder: null,
+                ref argsToParams,
+                out var defaultArguments,
+                expanded,
+                enableCallerInfo: true,
+                new DiagnosticBag());
```

Comment:
should we use a pooled diagnostic bag (even if the diagnostics will be discarded?)